File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/builders/skins/Skin.java
package cz.cvut.fel.matyapav.afandroid.builders.skins;

import android.graphics.Typeface;
import android.view.ViewGroup;

/**
 * Created by Pavel on 24.02.2016.
 */
public interface Skin {

    //common
    ViewGroup.LayoutParams getTopLayoutParams();

    int getComponentMarginLeft();
    int getComponentMarginRight();
    int getComponentMarginTop();
    int getComponentMarginBottom();

    //forms
    int getLabelColor();

    int getFieldColor();

    int getValidationColor();

    int getInputWidth();

    Typeface getValidationFont();

    Typeface getFieldFont();

    Typeface getLabelFont();

    int getLabelWidth();

    int getLabelHeight();

    //tables
    int getTableHeight();
    int getHeaderRowHeight();
    int getContentRowHeight();

    int getHeaderRowBackgroundColor();
    int getHeaderRowTextColor();
    int getContentBackgroundColor();
    int getContentTextColor();
    int getBorderColor();

    int getCellPaddingLeft();
    int getCellPaddingRight();
    int getCellPaddingTop();
    int getCellPaddingBottom();

    int getHeaderRowGravity();
    int getContentGravity();

    int getBorderWidth();

    //lists
    int getListWidth();
    int getListHeight();
    int getListItemBackgroundColor();
    int getListItemNameColor();
    int getListItemTextColor();
    Typeface getListItemNameFont();
    Typeface getListItemTextFont();
    int getListItemNameSize();
    int getListItemsTextSize();
    boolean isListItemNameLabelVisible();
    boolean isListItemTextLabelsVisible();
    boolean isListScrollBarAlwaysVisible();

    int getListItemTextPaddingLeft();
    int getListItemTextPaddingRight();
    int getListItemTextPaddingTop();
    int getListItemTextPaddingBottom();

    int getListItemNamePaddingLeft();

    int getListItemNamePaddingRight();

    int getListItemNamePaddingTop();

    int getListItemNamePaddingBottom();

    int getListContentWidth();

    int getListBorderColor();
    float getListBorderWidth();
}

File: AFMobile/AFWindowsPhone/AFWindowsPhone/builders/skins/Skin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.UI;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Media;

namespace AFWindowsPhone.builders.skins
{
    interface Skin
    {

        int getComponentMarginLeft();
        int getComponentMarginRight();
        int getComponentMarginTop();
        int getComponentMarginBottom();

        //forms
        Color getLabelColor();

        Color getFieldColor();

        Color getValidationColor();

        int getInputWidth();
        HorizontalAlignment getInputHorizontalAlignment();

        FontFamily getValidationFont();
        int getValidationFontSize();

        FontFamily getFieldFont();
        int getFieldFontSize();

        FontFamily getLabelFont();
        int getLabelFontSize();

        int getLabelWidth();
        HorizontalAlignment getLabelHorizontalAlignment();
        int getLabelHeight();
        VerticalAlignment getLabelVerticalAlignment();

        //lists
        int getListWidth();
        int getListHeight();

        HorizontalAlignment getListHorizontalAlignment();
        VerticalAlignment getListVerticalAlignment();
  
        Color getListItemBackgroundColor();
        Color getListItemNameColor();
        Color getListItemTextColor();
        FontFamily getListItemNameFont();
        FontFamily getListItemTextFont();
        int getListItemNameSize();
        int getListItemsTextSize();
        bool isListItemNameLabelVisible();
        bool isListItemTextLabelsVisible();

        bool isListScrollBarAlwaysVisible();

        int getListItemTextPaddingLeft();
        int getListItemTextPaddingRight();
        int getListItemTextPaddingTop();
        int getListItemTextPaddingBottom();

        int getListItemNamePaddingLeft();

        int getListItemNamePaddingRight();

        int getListItemNamePaddingTop();

        int getListItemNamePaddingBottom();

        int getListContentWidth();
        HorizontalAlignment getListContentHorizontalAlignment();

        Color getListBorderColor();
        float getListBorderWidth();
    }
}
